File: vmlib/src/main/java/me/shouheng/vmlib/anno/ViewModelConfiguration.java
package me.shouheng.vmlib.anno;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * VM 的配置类
 *
 * @author <a href="mailto:shouheng2015@gmail.com">WngShhng</a>
 * @version 2020-01-10 21:38
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
public @interface ViewModelConfiguration {

    /**
     * Will the event bus will be used in this vm.
     *
     * @return true if you want to use event bus.
     */
    boolean useEventBus() default false;
}

File: vmlib/src/main/java/me/shouheng/vmlib/anno/ViewModelConfiguration.kt
package me.shouheng.vmlib.anno

/**
 * ViewModel configuration annotation.
 *
 * @author [ShouhengWang](mailto:shouheng2020@gmail.com)
 * @version 2020-01-10 21:38
 */
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS)
annotation class ViewModelConfiguration(

    /** Will the event bus will be used in this vm.*/
    val useEventBus: Boolean = false
)